pkg/utils/dataset/lifecycle/schedule.go

```diff
@@ -65,114 +79,144 @@ func AssignDatasetToNodes(runtimeInfo base.RuntimeInfoInterface,
 		}
 	}
 
-	for _, node := range alreadySchedNodeList.Items {
-		currentScheduledNodes[node.Name] = node
-		log.Info("Node is already assigned", "node", node.Name, "dataset", dataset.Name)
-	}
-
+	// 2. When in fuse global mode, sort nodes by number of running workloads on it.
 	fuseGlobal, nodeSelector := runtimeInfo.GetFuseDeployMode()
-
-	pvcMountNodesMap, err := kubeclient.GetPvcMountNodes(runtimeClient, dataset.Name, dataset.Namespace)
-	if err != nil {
-		log.Error(err, "Failed to get PVC Mount Nodes, will treat every node as no PVC mount Pods")
-	}
-
 	var nodes []corev1.Node
 
 	if fuseGlobal {
+		pvcMountNodesMap, err := kubeclient.GetPvcMountNodes(runtimeClient, dataset.Name, dataset.Namespace)
+		if err != nil {
+			log.Error(err, "Failed to get PVC Mount Nodes, will treat every n as no PVC mount Pods")
+		}
 		nodes = sortNodesToBeScheduled(nodeList.Items, pvcMountNodesMap, nodeSelector)
 	} else {
 		nodes = nodeList.Items
 	}
 
-	// storageMap := tieredstore.GetLevelStorageMap(runtime)
-	for _, node := range nodes {
+	// 3. Pick nodes that is already scheduled with this runtime worker pod
+	for _, node := range alreadySchedNodeList.Items {
+		currentScheduledNodes[node.Name] = node
+		log.Info("Node is already assigned", "n", node.Name, "dataset", dataset.Name)
```

Comment:
what does `n` mean ?

Reply:
Thx. This is a typo and now is reverted.